docs/building-modules/module-interfaces.md

```diff
@@ -0,0 +1,93 @@
+- **Arguments: `codec` and `queryRoute`.** In addition to taking in the application `codec`, query command getters also take a `queryRoute` used to construct a path [Baseapp](../core/baseapp.md) uses to route the query in the application.
+- **Construct the command.** Read the [Cobra Documentation](https://github.com/spf13/cobra) for details on how to create commands.
+- **`RunE`.** The function should be specified as a `RunE` to allow for errors to be returned. This function encapsulates all of the logic to create a new query that is ready to be relayed to nodes.
+  + The function should first initialize a new [`CLIContext`](./query-lifecycle.md#clicontext) with the application `codec`.
+  + If applicable, the `CLIContext` is used to retrieve any parameters (e.g. the query originator's address to be used in the query) and marshal them with the query parameter type, in preparation to be relayed to a node.
+  + Use the `queryRoute` to construct a route Baseapp will use to route the query to the appropriate [querier](./querier.md). Module queries are `custom` type queries.  
+  + Call the `CLIContext` query function to relay the query to a node and retrieve the response.
+  + Unmarshal the response and use the `CLIContext` to print the output back to the user.
+- **Flags.** Add any [flags](#flags) to the command.
+
+
+Finally, the module also needs a `GetQueryCmd`, which aggregates all of the query commands of the module. Application developers wishing to include the module's queries will call this function to add them as subcommands in their CLI.
+
+### Flags
+
+[Flags](../interfaces/cli.md#flags) are entered by the user and allow for command customizations. Examples include the fees or gas prices users are willing to pay for their transactions.
+
+The flags for a module are typically found in the `flags.go` file in the `/client/cli` folder. Module developers can create a list of possible flags including the value type, default value, and a description displayed if the user uses a `help` command. In each transaction getter function, they can add flags to the commands and, optionally, mark flags as _required_ so that an error is thrown if the user does not provide values for them.
+
+For full details on flags, visit the [Cobra Documentation](https://github.com/spf13/cobra).
+
+## REST
+
+Application users may find the most intuitive methods of interfacing with the application are web services that use HTTP requests (e.g. a web wallet like [Lunie.io](lunie.io)). Thus, application developers will also use REST Routes to route HTTP requests to the application's modules. The module developer's responsibility is to define the REST client by defining routes for all possible requests and handlers for each of them. The REST interface file is typically found in the module's `/client/rest` folder.
```

Comment:
Hmm no, it's not because it's more intuitive. It's just a requirement for many service providers to have an interface that handles HTTP requests.

Reply:
application developers build the routes, service providers use them